spec/features/favorites/add_favorite_pet_spec.rb

```diff
@@ -21,8 +21,30 @@
 
       click_button("Add to Favorites")
 
+      expect(current_path).to eq("/pets/#{pet_2.id}")
       expect(page).to have_content("Pet has been added to your favorites list")
       expect(page).to have_content("Favorite Pets: 2")
     end
+
+    it "I cannot add a pet more than once" do
+      shelter = Shelter.create(name: "Happy Puppies", address: "55 Street St", city: "Danger Mountain", state: "UT", zip: "80304")
+      pet_1 = Pet.create(image: "image.jpeg", name: "Kunga", approximate_age: "1", sex: "male", shelter_id: shelter.id)
+
+      visit "/pets/#{pet_1.id}"
+
+      click_button("Add to Favorites")
+
+      expect(page).to have_content("Favorite Pets: 1")
+      expect(page).to_not have_content("Add to Favorites")
+      expect(page).to have_button("Remove from Favorites")
+
+      click_button("Remove from Favorites")
+
+      expect(current_path).to eq("/pets/#{pet_1.id}")
+      expect(page).to have_content("Pet has been removed from your favorites list")
+      expect(page).to_not have_content("Add to Favorites")
```

Comment:
In the user story it says that after removing a pet from favorites, when we go back to that pet's show page, "I can now see a link to favorite that pet." My understanding is that the button should come back after we remove that pet from the favorites. This seems like it should be an easy fix, right?